rubix-tests/src/test/java/com/qubole/rubix/tests/TestThriftServerJVM.java

```diff
@@ -0,0 +1,176 @@
+     * */
+    String[] bookKeeperStartCmd = {hadoopDirectory, "jar", bookKeeperJarPath, bookKeeperClass, setDataBlockSize, setCacheMaxDisks};
+    String[] localDataTransferStartCmd = {hadoopDirectory, "jar", bookKeeperJarPath, localDataTransferServerClass, setDataBlockSize, setCacheMaxDisks};
+
+    ProcessBuilder pJVMBuilder = new ProcessBuilder();
+    pJVMBuilder.redirectErrorStream(true);
+    pJVMBuilder.inheritIO();
+    pJVMBuilder.command(bookKeeperStartCmd);
+    bookKeeperJvm = pJVMBuilder.start();
+    pJVMBuilder.command(localDataTransferStartCmd);
+    localDataTransferJvm = pJVMBuilder.start();
+
+    log.info("Test Directory : " + testDirectory);
+    Files.createDirectories(Paths.get(testDirectory));
+    Thread.sleep(3000);
+  }
+
+  @AfterClass
+  public static void tearDownClass() throws IOException
+  {
+    /* ****
+      Destroying bookKeeper and localDataTransfer JVMs
+    **** */
+    bookKeeperJvm.destroy();
+    localDataTransferJvm.destroy();
+
+    log.info("Deleting files in " + testDirectory);
+    Files.walkFileTree(Paths.get(testDirectory), new DeleteFileVisitor());
+    Files.deleteIfExists(Paths.get(testDirectory));
+
+    Files.walkFileTree(Paths.get(cacheDir + "/" + testDirectoryPrefix), new DeleteFileVisitor());
+    Files.deleteIfExists(Paths.get(cacheDir + "/" + testDirectoryPrefix));
+  }
+
+  @BeforeMethod
+  public static void setup() throws IOException, InterruptedException, URISyntaxException
+  {
+    DataGen.populateFile(backendFileName);
+    log.info("BackendPath: " + backendPath);
+
+    CacheConfig.setIsStrictMode(conf, true);
+    CacheConfig.setCacheDataDirPrefix(conf, testDirectoryPrefix + "dir");
+    CacheConfig.setMaxDisks(conf, 2);
+    CacheConfig.setIsParallelWarmupEnabled(conf, false);
+    CacheConfig.setBlockSize(conf, 200);
+  }
+
+  @AfterMethod
+  public static void cleanup()
+  {
+    File file = new File(backendFileName);
+    file.delete();
+  }
+
+  @Test
+  public void testJVMCommunication() throws IOException, InterruptedException
+  {
+    log.info("Value of Path " + this.backendFileName);
+    log.debug(" backendPath to string : " + this.backendPath.toString());
+
+    String host = "localhost";
+    boolean dataDownloaded;
+    File file = new File(backendFileName);
+    List<BlockLocation> result;
+    int lastBlock = 4;
+    int readSize = 1000;
+
+    try {
+      client = bookKeeperFactory.createBookKeeperClient(host, conf);
+
+      result = client.getCacheStatus("file:///" + backendFileName, file.length(), file.lastModified(), 0, lastBlock, 3);
+      assertTrue(result.get(0).getLocation() == Location.LOCAL, "File already cached, before readData call");
+      log.info(" Value of Result : " + result);
+      log.info("Downloading file from path : " + file.toString());
+      dataDownloaded = client.readData("file:///" + backendFileName, 0, readSize, file.length(), file.lastModified(), 3);
+      if (!dataDownloaded) {
```

Comment:
The test should not swallow this. This seems like a condition on which the test should fail.

Reply:
We should fail the test if dataDownloaded is false. It means we have not been able to download the file properly.